### PR TITLE
[UXE-4473] fix: block users without permission of acessing team permissions edit page

### PR DESCRIPTION
**Pull request by aloisio-m-bastian**
### Explain what was fixed and the correct behavior.
Block users withous permission of acessing team permissions edit page
### Does this PR introduce UI changes? Add a video or screenshots here.

https://github.com/user-attachments/assets/ee67cef1-c50d-4f4b-8924-b3880f7267ac

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] bug: TITLE
- [x] Commits are tagged with the right word (feat, test, refactor, etc)
- [ ] Application responsiveness was tested to different screen sizes
- [x] New tests are added to prevent the same issue from happening again
- [ ] UI changes are validated by a team designer
- [x] Code is formatted and linted
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

File: src/views/TeamsPermissions/EditView.vue [hunk 1]
```diff
@@ -54,6 +56,13 @@
       required: true
     }
   })
+
+  const router = useRouter()
+
+  const handleLoadFail = () => {
+    router.push({ name: 'teams-permission' })
```
Comment (matheus-sunderhus) [inline src/views/TeamsPermissions/EditView.vue:63]: add an E2e test cenário, where this user, without the permission will be pushed to  teams-permission route  when Edit fails.

You can create a e2e test, and intercept de load service to create a error and ensure this behavior

**Review by matheus-sunderhus — DISMISSED** [review]
LGTM

**Review by matheus-sunderhus — COMMENTED** [review]
e2e tests required to ensure this behavior

**Review by pauloSF0 — APPROVED** [review]

**Review by pauloSF0 — APPROVED** [review]

File: cypress/e2mock/teams-permissions.cy.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    detail: 'You do not have permission to perform this action (teams management).'
+  }
+}
+describe('Tams Permissions Spec', () => {
+  beforeEach(() => {
+    cy.login()
+    cy.openProduct('Teams Permissions')
+  })
+
+  it('should block users without acess of entering in the edit page', function () {
```
Comment (matheus-sunderhus) [inline cypress/e2mock/teams-permissions.cy.js:14]: typo on title description

File: cypress/e2mock/teams-permissions.cy.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    //act
+    cy.get(selectors.list.searchField).type('Default Team')
+    cy.intercept('GET', '/api/v4/iam/teams/*', { statusCode: 403, body: fixtures.errorMessage }).as(
```
Comment (matheus-sunderhus) [inline cypress/e2mock/teams-permissions.cy.js:17]: use cy.intercept, using only the specific  service that needs to be intercepted cypress in the teams collection.

**Review by matheus-sunderhus — COMMENTED** [review]
minor changes

File: cypress/e2mock/teams-permissions.cy.js (hunk 3 above)
Comment (aloisio-m-bastian) [reply cypress/e2mock/teams-permissions.cy.js:17]: o asterisco na rota do intercept foi utilizado para que não seja necessário especificar o id do team que esta sendo requisitado

**Review by matheus-sunderhus — APPROVED** [review]
lgtm

**Review by pauloSF0 — APPROVED** [review]

**Pull request merged**